app/assets/src/components/views/SampleView/SampleView.jsx

```diff
@@ -404,11 +413,54 @@ class SampleView extends React.Component {
       return {};
     }
 
+    let accessionData = null;
+
+    // For genus-level taxons, we aggregate all the available species-level taxons for that genus.
+    if (coverageVizParams.taxLevel === "genus") {
```

Comment:
For code clarity, put this whole block in a method `getCoverageVizParamsForGenus` or similar